src/Support/Authentication.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Nuwave\Lighthouse\Support;
+
+class Authentication
+{
+    /**
+     * @return string|null
+     */
+    public static function getGuard()
+    {
+        $guards = array_keys(config('auth.guards'));
+
+        if (! empty($guards)) {
+            foreach ($guards as $guard) {
```

Comment:
That just checks all guards that exist, disregarding if they are even configured to be used. I would not expect this behaviour as a user, that makes it very dangerous.

This also completely disregards the setting in `lighthouse.guard`, that is what I meant my comment here: https://github.com/nuwave/lighthouse/pull/1630#discussion_r534194985